pkg/alerts/alertsqlite/alerts_sqlite.go

```diff
@@ -625,49 +655,73 @@ func (p Sqlite) DeleteContactPoint(contact_id string) error {
 }
 
 // update last_sent_time and last_alert_state in notification_details table
-func (p Sqlite) UpdateLastSentTimeAndAlertState(alert_id string, alertState alertutils.AlertState) error {
+func updateLastSentTimeAndAlertState(tx *gorm.DB, alert_id string, alertState alertutils.AlertState) error {
 	currentTime := time.Now().UTC()
 
-	if err := p.db.Model(&alertutils.Notification{}).Where("alert_id = ?", alert_id).
+	if err := tx.Model(&alertutils.Notification{}).Where("alert_id = ?", alert_id).
 		Updates(map[string]interface{}{
 			"last_sent_time":   currentTime,
 			"last_alert_state": alertState,
 		}).Error; err != nil {
 		err = fmt.Errorf("UpdateLastSentTimeAndAlertState: unable to update, AlertId=%v, Error=%+v", alert_id, err)
-		log.Errorf(err.Error())
 		return err
 	}
 
 	return nil
 }
 
-func (p Sqlite) UpdateAlertStateAndIncrementNumEvaluations(alert_id string, alertState alertutils.AlertState) error {
+func updateAlertStateAndIncrementNumEvaluations(tx *gorm.DB, alert_id string, alertState alertutils.AlertState) error {
+	if err := tx.Model(&alertutils.AlertDetails{}).Where("alert_id = ?", alert_id).
+		Updates(map[string]interface{}{
+			"state":                 alertState,
+			"num_evaluations_count": gorm.Expr("num_evaluations_count + ?", 1),
+		}).Error; err != nil {
+		err = fmt.Errorf("UpdateAlertStateAndIncrementNumEvaluations: unable to update alert state and increment evaluations count with AlertId=%v, Error=%+v", alert_id, err)
+		return err
+	}
```

Comment:
nit: I think this syntax you're using:
```
if err := foo(); err != nil {
    // handle error
}
```
is fine when the whole condition is on one line, but since your condition spans several lines it gets a bit harder to read. Can you do it this way instead?
```
err := foo
if err != nil {
    // handle error
}
```